public/pages/SplitIndex/container/SplitIndex/SplitIndex.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { Component } from "react";
+import { EuiSpacer, EuiTitle, EuiButton } from "@elastic/eui";
+import { get } from "lodash";
+
+import { CatIndex } from "../../../../../server/models/interfaces";
+import { BrowserServices, RecoveryJobMetaData } from "../../../../models/interfaces";
+import SplitIndexFlyout from "../../components/SplitIndexFlyout";
+import { IndexItem } from "../../../../../models/interfaces";
+import { RouteComponentProps } from "react-router-dom";
+import { jobSchedulerInstance } from "../../../../context/JobSchedulerContext";
+import queryString from "query-string";
+import {
+  openIndices,
+  getIndexSettings,
+  setIndexSettings,
+  getSplitShardOptions,
+  splitIndex,
+  getSingleIndice,
+  getAlias,
+} from "../../../Indices/utils/helpers";
+
+import { CommonService, ServicesContext } from "../../../../services";
+import { CoreStart } from "opensearch-dashboards/public";
+import { useContext } from "react";
+import { CoreServicesContext } from "../../../../components/core_services";
+import { ROUTES } from "../../../../utils/constants";
+
+interface SplitIndexProps extends RouteComponentProps {
+  commonService: CommonService;
+  coreService: CoreStart;
+}
+
+export class SplitIndex extends Component<SplitIndexProps> {
+  state = {
+    reasons: [] as React.ReactChild[],
+    shardsSelectOptions: [] as { label: string }[],
+    sourceIndex: {} as CatIndex,
+    splitIndexFlyoutVisible: false,
+  };
+
+  async componentDidMount() {
+    await this.isSourceIndexReady();
+    this.calculateShardsOption();
+    this.setState({
+      splitIndexFlyoutVisible: true,
+    });
+  }
+
+  isSourceIndexReady = async () => {
+    const source = queryString.parse(this.props.location.search) as { source?: string };
```

Comment:
what's the behavior when query string don't have source parameter? let's say user type the URL directly in browser.

Reply:
User will get a blank page and an error saying that not able to get index setting